app/jobs/issue_event_job.rb

```diff
@@ -34,30 +41,37 @@ def self.build_meta(params)
   end
 
   def guard_against_double_events
-    payload = { meta:  self.class.build_meta(self.arguments.first) }
+    message = build_message(self.arguments.first)
     willPublish = true
+
     Rails.cache.with do |dalli|
-      key = "#{payload[:meta][:action]}.#{payload[:meta][:user]["login"]}.#{payload[:meta][:identifier]}.#{payload[:meta][:timestamp]}"
+      key = self._cache_key.call(message)
       willPublish = false if dalli.get(key)
-      dalli.set(key, payload.to_s)
+      dalli.set(key, key)
     end
+
     yield if willPublish
   end
 
+  def build_message(params)
+    payload = payload(params)
+    payload['actor'] = self.arguments.first['current_user']
+
+    message = HashWithIndifferentAccess.new({
```

Comment:
Precautionary at this point. 

The code base has too many places scattered around that switch between symbol and string access